src/ray/rpc/server_call.h

```diff
@@ -161,6 +175,7 @@ class ServerCallImpl : public ServerCall {
         response_writer_(&context_),
         io_service_(io_service),
         call_name_(std::move(call_name)),
+        cluster_id_(cluster_id),
```

Comment:
This line has a different semantics when you change from a pointer to a reference. This uses a copy constructor which is defined in id.h and creates a new object.